main.js

```diff
@@ -124,20 +159,21 @@ if (config.downloadJAD) {
   loadingMIDletPromises.push(new Promise(function(resolve, reject) {
     JARStore.loadJAR("midlet.jar").then(function(loaded) {
       if (loaded) {
+        if (!document.hidden) {
+          // Show the splash screen as soon as possible.
+          showSplashScreen();
```

Comment:
> I guess there's no way to show this earlier, because we don't know if we need to download the MIDlet or not.

Right. Initially, I showed it as soon as possible, before knowing about the download. But that created a strange effect on install run, as the app first showed "Starting…", then "Downloading…", then "Starting…" again.